build.gradle

```diff
@@ -56,6 +61,26 @@ if (project.hasProperty("maxParallelForks")) {
   ext.maxParallelForks = 3
 }
 
+animalsniffer {
+  sourceSets = [sourceSets.main]
+}
+
+findbugs {
+  // Some errors are logged due to https://github.com/gradle/gradle/issues/1094, but Findbugs checks are working fine
+  toolVersion '3.0.0'
+  sourceSets = [sourceSets.main]
+}
+
+codenarc {
+  // Some errors are logged due to https://github.com/gradle/gradle/issues/1094, but Findbugs checks are working fine
+  toolVersion '0.27.0'
+  sourceSets = [sourceSets.test]
+}
+
+configurations {
+  // https://github.com/gradle/gradle/issues/1305
+  compile.exclude group: 'com.ibm.icu', module: 'icu4j'
```

Comment:
Nothing extraordinary. Just declares the dependency and uses this configuration:
`<plugin>
        <groupId>org.codehaus.mojo</groupId>
        <artifactId>animal-sniffer-maven-plugin</artifactId>
        <executions>
          <execution>
            <goals>
              <goal>check</goal>
            </goals>
            <id>check</id>
          </execution>
        </executions>
        <configuration>
          <signature>
            <groupId>org.codehaus.mojo.signature</groupId>
            <artifactId>java1${java.level}</artifactId>
          </signature>
        </configuration>`

As far as I know, the HPI plugin does not include either the core (as it is a provided dependency) nor other plugins dependencies (as they are handled by the plugin manager).